.github/workflows/terraform.yml

```diff
@@ -70,6 +72,32 @@ jobs:
           soft_fail: true
           working_directory: "${{ matrix.module }}"
 
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.4
+        env:
+          GO111MODULE: "on"
+
+      - name: Install Modules
+        working-directory: "modules"
+        env:
+          GO111MODULE: "on"
+        run: |
+          go mod download
+          go mod tidy
+
+      - name: Terraform Terratest
+        working-directory: "${{ matrix.module }}/test"
```

Comment:
I expect this will fail when test directory does not exist for that module - it's passing here because there are no changes in other modules

Reply:
Yah I'd assume so, I'll have a think about it if I give this a proper go.